### PR TITLE
✨[Story layout] Added validation to presets

### LINKED ISSUE #31284: [Story content layout] Write validation rules for presets on grid-layer
Write validation rules for `amp-story-grid-layer[preset]`.

cc @ampproject/wg-stories

### PR DESCRIPTION
**Pull request by mszylkowski**
Created validation rules and tests for presets and anchoring. scaling-factor is not valid.

Closes #31284

### CODE REVIEW

File: extensions/amp-story/1.0/test/validator-amp-story-grid-layer-presets.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<!--
+  Copyright 2019 The AMP HTML Authors. All Rights Reserved.
```
Comment (honeybadgerdontcare) [inline extensions/amp-story/1.0/test/validator-amp-story-grid-layer-presets.html:2]: s/2019/2021

File: extensions/amp-story/validator-amp-story.protoascii [hunk 2]
```diff
@@ -173,6 +173,15 @@ tags: {  # <amp-story-grid-layer>
     name: "aspect-ratio"
     value_regex: "\\d+:\\d+"
   }
+  attrs: {
```
Comment (honeybadgerdontcare) [inline extensions/amp-story/validator-amp-story.protoascii:176]: please alphabetize attrs by name

**Review by honeybadgerdontcare — CHANGES_REQUESTED** [review]
Why `2021-*`? Is there an issue this can be linked to? Thanks!

**Comment by mszylkowski** [comment]
The recommended values could change if mobile devices change screen sizes drastically (not expected to happen anytime soon), so the year serves as a versioning system to ensure we can change the recommendation if necessary by pushing a new set of presets

File: extensions/amp-story/validator-amp-story.protoascii [hunk 3]
```diff
@@ -173,6 +173,15 @@ tags: {  # <amp-story-grid-layer>
...
+    name: "preset"
+    value: "2021-background"
+    value: "2021-foreground"
+  }
+  attrs: {
+    name: "anchor"
+    value_regex: "top|bottom|left|right|(top|bottom) (left|right)"
```
Comment (gmajoulet) [inline extensions/amp-story/validator-amp-story.protoascii:183]: We should allow:
top
left
top-left
top left
left top
left-top

File: extensions/amp-story/1.0/test/validator-amp-story-grid-layer-presets.out [hunk 4]
```diff
@@ -0,0 +1,54 @@
+PASS
+|  <!--
+|    Copyright 2019 The AMP HTML Authors. All Rights Reserved.
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/test/validator-amp-story-grid-layer-presets.out:3]:
```suggestion
|    Copyright 2021 The AMP HTML Authors. All Rights Reserved.
```

**Review by Gregable — APPROVED** [review]

**Review by gmajoulet — APPROVED** [review]

File: extensions/amp-story/1.0/test/validator-amp-story-grid-layer-presets.out (hunk 4 above)
Comment (honeybadgerdontcare) [reply extensions/amp-story/1.0/test/validator-amp-story-grid-layer-presets.out:3]: Might want to run `gulp validator --update-tests` to get the new date automatically and verify the changes to the protoascii are reflected here.

**Review by honeybadgerdontcare — APPROVED** [review]

**Pull request merged**